zerver/views/streams.py

```diff
@@ -432,7 +433,9 @@ def add_subscriptions_backend(
             )
 
     if len(notifications) > 0:
-        do_send_messages(notifications)
+        message_ids = do_send_messages(notifications)
```

Comment:
I think this will catch too many messages? E.g. the ones telling other users that they were subscribed.

Reply:
What `do_send_messages` will do is that it will return a list of all of the new *message* ids. Now what we do in `mark_messages_as_read_by_stream_creator` is that we filter *usermessages* based on these message ids and the user_profile of the user who contacted the endpoint (and thus created the stream). This way, we're not marking any other users' messages as read. Also, the test added here verifies that this is indeed true. :slightly_smiling_face:

Reply:
ah, makes sense.